core/sql/ustat/hs_log.h

```diff
@@ -277,5 +283,8 @@ class HSLogMan
          timeval prevTime_;
          char title_[MAX_TIMING_EVENTS][1000];
          static THREAD_P HSLogMan* instance_;      /* 1 and only 1 instance           */
+
+         // intended as a long-term replacement for msg + Log()
+         NAString logStreamBuffer_;
```

Comment:
I like this change very much. Using an NAString seems better than using fixed length buffers. I do wonder if this NAString needs to be allocated on a partcular heap. I think the default, is to use system heap. Since this class is a singleton maybe the context heap is a better place for it, allowing heap usage to traced using executor counters. I am not sure if the heap change is necessary or even beneficial when considering all aspects such as multiple Contexts being present in situations such as T2 and repository threads. Thread safety of this variable when we have multiple T2 connections doing update stats on different tables from the same process may also be an interesting test. These are minor concerns. The approach and change are a resounding +1 for me.

Reply:
Thanks for the review. I'll explore putting logStreamBuffer_ on an NAHeap in a later pull request. One bit of good news is that this code runs in the tdm_arkcmp process, so we won't be affecting global heap in mxosrvr.